Quality_Declaration_foonathan_memory.md

```diff
@@ -0,0 +1,206 @@
+# **External Dependency Quality Declaration** `foonathan_memory`
+
+This document is a declaration of software quality for the `foonathan_memory` external dependency, inspired on the
+guidelines provided in the [ROS2 REP-2004 document](https://ros.org/reps/rep-2004.html).
+
+Foonathan's [`memory`](https://github.com/foonathan/memory) is a C++ library to manage memory allocations that improves upon the STL.
+*eprosima Fast DDS* started using this library following the advice of ROS's and eProsima's partner, Apex.AI.
+
+Even though several requirements stablished in REP-2004 are not met and, consequently, `foonathan_memory` cannot claim a high Quality Level, this document assess the quality risks that can be introduced by the use of this library in *eprosima Fast DDS*.
+This Quality Declaration claims that the external dependency `foonathan_memory` qualifies to Quality Level 2 category for its use within *eprosima Fast DDS*.
+
+Below are the rationales, notes and caveats for this claim, organized by the requirements listed in the [Package Requirements for Quality Level 1 in REP-2004](https://www.ros.org/reps/rep-2004.html#package-requirements).
+
+##  Version Policy [1]
+
+### Version Scheme [1.i]
+
+
+`foonathan_memory` does not have a declared versioning scheme.
+
+The latest release can be found [here](https://github.com/foonathan/memory/releases) and the release notes can be found in the [CHANGELOG](CHANGELOG.MD).
+Even though the current version is `< 1.0.0`, `foonathan_memory` is in [maintenance mode](https://www.jonathanmueller.dev/project/).
+
+*eprosima Fast DDS* ensures `foonathan_memory` version stability by pinning to a specific [commit](https://github.com/foonathan/memory/pull/70).
```

Comment:
Maybe link commit instead of pull request?
https://github.com/foonathan/memory/commits/41808c3d348a1ee2e040775085e5e43d9bc69f32